build/commands/lib/test.js

```diff
@@ -173,4 +182,55 @@ const runTests = (passthroughArgs, suite, buildConfig, options) => {
   }
 }
 
+const runChromiumTestLauncherTeamcityReporterIntegrationTest = () => {
+  const args = [
+    "--gtest_filter=DISABLED_TeamcityReporterIntegration*",
+    "--gtest_also_run_disabled_tests",
+  ]
+
+  const runOptions = config.defaultOptions
+  runOptions.stdio = 'pipe'
+  runOptions.continueOnFail = true
+  const prog = util.run(path.join(config.outputDir, 'brave_unit_tests'), args, runOptions)
+
+  const expectedOutput = [
+    "##teamcity[testSuiteStarted name='brave_unit_tests']",
+    "##teamcity[testRetrySupport enabled='false']",
+    "##teamcity[testStarted name='DISABLED_TeamcityReporterIntegrationTest.Success' captureStandardOutput='true']",
+    "##teamcity[testFinished name='DISABLED_TeamcityReporterIntegrationTest.Success' duration='",
+    "##teamcity[testStarted name='DISABLED_TeamcityReporterIntegrationTest.Failure' captureStandardOutput='true']",
+    "##teamcity[testFailed name='DISABLED_TeamcityReporterIntegrationTest.Failure']",
+    "##teamcity[testFinished name='DISABLED_TeamcityReporterIntegrationTest.Failure' duration='",
+    "##teamcity[testStarted name='DISABLED_TeamcityReporterIntegrationTest.CheckFailure' captureStandardOutput='true']",
+    "##teamcity[testFailed name='DISABLED_TeamcityReporterIntegrationTest.CheckFailure']",
+    "##teamcity[testFinished name='DISABLED_TeamcityReporterIntegrationTest.CheckFailure' duration='",
+    "##teamcity[testStarted name='DISABLED_TeamcityReporterIntegrationTest.Skipped' captureStandardOutput='true']",
+    "##teamcity[testIgnored name='DISABLED_TeamcityReporterIntegrationTest.Skipped']",
+    "##teamcity[testFinished name='DISABLED_TeamcityReporterIntegrationTest.Skipped' duration='",
```

Comment:
Do we need `testFinished` with `testIgnored`?
from the specs:
```As an exception, the testIgnored message can be reported without the matching testStarted and testFinished messages.```

Reply:
technically we do not need `testStarted/testFinished` in this case.

Initially I did exactly this (didn't generate these events in `SKIPPED` state), but it added unnecessary conditions into `OnTestStarted` and `OnTestFinished` which I decided to drop to have a simpler codebase.